liberty-maven-plugin/src/it/dev-it/src/test/java/net/wasdev/wlp/test/dev/it/RunTest.java

```diff
@@ -57,6 +57,9 @@ public void endpointTest() throws Exception {
             String response = method.getResponseBodyAsString();
 
             assertTrue("Unexpected response body", response.contains("hello world"));
+            // Note that checkLogMessage returns true if it does NOT find the message.
+            // So I am verifying that SLF4J loaded properly.
+            assertTrue(checkLogMessage(2000, "SLF4J: Failed to load class"));
```

Comment:
Why did you decide the 2000? I'm assuming that's some sort of time limit so if you don't find that message within 2 seconds then the test passes?

Can we test that the class loaded successfully in a more explicit manner, such as by looking for the `Here is the Log` message that is logged?

Reply:
Before making the change to the resolution scope, this error message showed up without fail. I did not see any different message I could look for in the affirmative case after the change was made. I do not know where the "Here is the log" message is going. I don't remember seeing it in the messages.log.

Reply:
The `BaseDevTest` class is what is providing the setup/teardown for the logging process. That is also where `checkLogMessage` is implemented.

The other tests that use `checkLogMessage` assert false, which means that the message must have been found within the given amount of time. On the other hand, this test is asserting true, which means the message was not found in that amount of time. That could mean one of two things: it's not ever going to be logged (which is what we want to confirm), or the timeout was too short and it will be logged later (which results in a false positive). 

All this goes back to wanting to check that the message that is logged is there. I'm not experienced in working with Java loggers, but a quick search indicates that slf4j provides the API and logback-classic is the underlying implementation that must be configured. [This article](https://mkyong.com/logging/logback-xml-example/) shows an example configuration which is supposed to log the output to the console. In this case it should be picked up the `checkLogMessage` method as the dev test setup redirects console output of the dev mode process to the log file that `checkLogMessage` is checking.

Reply:
I understand your point, and I could increase the time if you would like. But the message would definitely be there if the logging implementation classes were not found (as was the case when the dependency was not included before this fix). As soon as the servlet is invoked it tries to access the logger and generates that message. So checking the log in the test case to make sure that message is not there after invoking the servlet is valid and is confirming that the fix worked.

Reply:
Ok, we can keep as is then.

Reply:
I updated the test to also look for the logger message.